pkg/cmd/experimental/bundlesecret/bundle-secret.go

```diff
@@ -0,0 +1,47 @@
+package bundlesecret
+
+import (
+	"encoding/base64"
+	"flag"
+	"fmt"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	"github.com/spf13/cobra"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+)
+
+// NewCmdBundleSecret creates a new bunde-secret command. The bundle-secret command will walk a given directory
+// and load its filenames into kubernetes secret keys with values that are base64 encoded
+func NewCmdBundleSecret(f *clientcmd.Factory, parentName, name string, out io.Writer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "bundle-secret",
+		Short: "bundle data from an OpenShift dir to load kubernetes secret",
+		Long:  "bundle data from an OpenShift dir to load kubernetes secret",
+		Run: func(cmd *cobra.Command, args []string) {
+			//need to handle errors with certidir
+			certdir := flag.Arg(0)
+			filepath.Walk(certdir, walkpath)
+		},
+	}
+	return cmd
+}
+
+func walkpath(path string, info os.FileInfo, err error) error {
+	if !info.IsDir() {
+		if util.IsDNS1123Subdomain(path) {
+			bytes, err := ioutil.ReadFile(path)
+			if err != nil {
+				fmt.Errorf("Could not read file %v", path)
+			}
+			encodedBytes := base64.StdEncoding.EncodeToString(bytes)
+			secret := make(map[string]string)
```

Comment:
I am sure you know this already, but you will need to structure this code differently so each file you see in a walkPath call can add its secret to a shared Secret object that we ultimately need to pipe out stdout